packages/web-components/src/utils/element-internals.ts

```diff
@@ -33,3 +33,36 @@ export function toggleState(elementInternals: ElementInternals, state: string, f
 
   elementInternals.states.delete(state);
 }
+
+/**
+ * A decorator for toggling the element internals of an element
+ *
+ * @internal
+ */
+export function toggleAttrState(target: any, attr: string): any {
```

Comment:
Love all the reduced code! That's a lot of lines of code! 💯 

One of my concerns is that this puts all the `...Changed` methods as public. But given most of the current ones are public anyways, maybe it's not a big deal. The other is that this removes all the `...Changed` methods from the API Report. I'm not sure how it's been used so can't comment on if it's something we need to fix but wanted to point it out in case it matters.

Reply:
I had the same question about `...Changed` being removed from the API Report. I could be convinced either way, but to me managing state-based styling hooks was overloading/bloating our API surface area and this makes those class functions a lot more succinct and exact about the work being done.